tests/infrastructure/utils.py

```diff
@@ -0,0 +1,57 @@
+import logging
+
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.deployment import Deployment
+from ocp_resources.kubelet_config import KubeletConfig
+
+from utilities.exceptions import ResourceMissingFieldError, ResourceValueError
+
+LOGGER = logging.getLogger(__name__)
+
+
+def verify_tekton_operator_installed(client: DynamicClient) -> None:
+    """Verify Tekton operator is installed and available.
+
+    Args:
+        client: pass client argument
+
+    Raises:
+        ResourceNotFoundError: If Tekton operator deployment is not found.
+        TimeoutExpiredError: If Tekton operator is not ready within the timeout period.
+    """
+    LOGGER.info("Verifying Tekton operator is installed and available")
+    tekton_deployment = Deployment(
+        name="openshift-pipelines-operator",
+        namespace="openshift-operators",
+        client=client,
+        ensure_exists=True,
+    )
+    tekton_deployment.wait_for_replicas()
+
+
+def verify_numa_enabled(client: DynamicClient) -> None:
+    """Verify cluster has nodes with NUMA topology and static CPU manager policy.
+
+    Args:
+        client: pass client argument
+
+    Raises:
+        ResourceMissingFieldError: If required fields are missing.
+        ResourceValueError: If cpuManagerPolicy has wrong value.
+    """
+    LOGGER.info("Verifying cluster has nodes with NUMA topology and static CPU manager policy")
+    for config in KubeletConfig.get(client=client):
```

Comment:
if there's only one `There is just one kubeletconfig available for cluster.` -why use `get`?